### PR TITLE
Revise admonition about provisioning profiles.

### PR DESCRIPTION
**Pull request by bbewan**
https://app.clubhouse.io/buddybuild/story/6926/add-note-that-for-ios-builds-deployments-to-devices-or-the-app-store-need-to-use-a-provisioning-profile-associated-with-an

### CODE REVIEW

**Comment by bbdocs** [comment]
See the website here: http://122-bb-docs.surge.sh

File: _common/important-provisioning_profiles.adoc [hunk 1]
```diff
@@ -4,7 +4,11 @@ After an app has been built, it is not possible to alter, remove, or
...
-If your app is built with a development provisioning profile, you
-cannot deploy your app to the App Store without rebuilding your app with
-a distribution provisioning profile.
+Xcode lets you create development provisioning profiles on your local
+computer, for deploying apps to connected devices. However, you cannot
+use a development provisioning profile to deploy your app to the App
+Store or to other devices. To deploy to the App Store, or to testers,
+you need to create a distribution provisioning profile in the Apple
+Developer Portal, which requires an active Developer Program
+subscription.
```
Comment (christopherstott) [inline _common/important-provisioning_profiles.adoc]: There's not a specific thing called a distribution provisioning profile. 

There are 4 kinds of provisioning profiles
1. Development provisioning profile (needs a developer code signing identity)
2. In House provisioning profile (needs a distribution code signing identity, and is only available on Enterprise teams)
3. Adhoc provisioning profile (needs a distribution code signing identity)
4. App Store provisioning profile (needs a distribution code signing identity)

I would say something like

```
To upload your app to the App Store, the app must be signed with an App Store provisioning profile. If you have your code signing type set to 'iOS Distribution' and an Apple Developer Portal Connection, buddybuild will automatically generate App Store provisioning profiles for you, and use those to produce a version of the app compatible with he App Store.

Getting App Store signed builds requires
- An 'iPhone Distribution' code signing identity to be present in buddybuild for the correct Apple Developer Team
- 'iPhone Distribution' set in code signing type
- An Apple Developer Portal Connection for the same Apple Developer Team.

If for some reason you are not able to add an Apple Developer Portal Connection, please contact our support team.
````

Comment (bbewan) [reply _common/important-provisioning_profiles.adoc]: Thanks for this, @christopherstott . With some minor tweaks, I've applied your wording.

**Comment by bbdocs** [comment]
See the website here: http://122-bb-docs.surge.sh

**Comment by bbdocs** [comment]
See the website here: http://122-bb-docs.surge.sh

**Comment by bbdocs** [comment]
See the website here: http://122-bb-docs.surge.sh

**Pull request merged**